src/code/z_collision_check.c

```diff
@@ -1,392 +1,3345 @@
+    D_8015D3D8.colorEnd[2].r = (u8)0x32;
+    D_8015D3D8.colorEnd[2].g = (u8)0x32;
+    D_8015D3D8.colorEnd[2].b = (u8)0x32;
+    D_8015D3D8.colorEnd[2].a = (u8)0x32;
+    D_8015D3D8.colorEnd[3].r = (u8)0;
+    D_8015D3D8.colorEnd[3].g = (u8)0;
+    D_8015D3D8.colorEnd[3].b = (u8)0;
+    D_8015D3D8.colorEnd[3].a = (u8)0;
+    D_8015D3D8.age = 0;
+    D_8015D3D8.duration = 16;
+    D_8015D3D8.velocity = 8.0f;
+    D_8015D3D8.gravity = -1.0f;
+    Effect_Add(globalCtx, &sp24, 0, 0, 1, &D_8015D3D8);
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_80062B80.s")
+#endif // NON_MATCHING
+
+void func_80062CD4(GlobalContext* globalCtx, Vec3f* v) {
+    s32 sp24;
+    static EffShieldParticleInit init = {
+        16,
+        { 0, 0, 0 },
+        { { 0x00, 0xC8, 0xFF, 0xFF } },
+        { { 0xFF, 0xFF, 0xFF, 0xFF } },
+        { { 0xFF, 0xFF, 0x80, 0xFF } },
+        { { 0xFF, 0xFF, 0x00, 0xFF } },
+        { { 0xFF, 0x40, 0x00, 0xC8 } },
+        { { 0xFF, 0x00, 0x00, 0xFF } },
+        2.1f,
+        35.0f,
+        30.0f,
+        8,
+        { 0, 0, 0, 0x00, 0x80, 0xFF, 0x00, 300 },
+        1,
+    };
+
+    init.position.x = (s32)v->x;
+    init.position.y = (s32)v->y;
+    init.position.z = (s32)v->z;
+    init.lightParams.posX = init.position.x;
+    init.lightParams.posY = init.position.y;
+    init.lightParams.posZ = init.position.z;
+    Effect_Add(globalCtx, &sp24, 3, 0, 1, &init);
+}
+
+void func_80062D60(GlobalContext* globalCtx, Vec3f* v) {
+    func_80062CD4(globalCtx, v);
+    Audio_PlaySoundGeneral(NA_SE_IT_SHIELD_REFLECT_SW, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+}
+
+void func_80062DAC(GlobalContext* globalCtx, Vec3f* v, Vec3f* arg2) {
+    func_80062CD4(globalCtx, v);
+    Audio_PlaySoundGeneral(NA_SE_IT_SHIELD_REFLECT_SW, arg2, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+}
+
+void func_80062DF4(GlobalContext* globalCtx, Vec3f* v) {
+    func_80062D60(globalCtx, v);
+}
+
+void func_80062E14(GlobalContext* globalCtx, Vec3f* v, Vec3f* arg2) {
+    s32 sp24;
+    static EffShieldParticleInit init = {
+        16,
+        { 0, 0, 0 },
+        { { 0x00, 0xC8, 0xFF, 0xFF } },
+        { { 0xFF, 0xFF, 0xFF, 0xFF } },
+        { { 0xFF, 0xFF, 0x80, 0xFF } },
+        { { 0xFF, 0xFF, 0x00, 0xFF } },
+        { { 0xFF, 0x40, 0x00, 0xC8 } },
+        { { 0xFF, 0x00, 0x00, 0xFF } },
+        2.1f,
+        35.0f,
+        30.0f,
+        8,
+        { 0, 0, 0, 0x00, 0x80, 0xFF, 0x00, 300 },
+        0,
+    };
+
+    init.position.x = (s32)v->x;
+    init.position.y = (s32)v->y;
+    init.position.z = (s32)v->z;
+    init.lightParams.posX = init.position.x;
+    init.lightParams.posY = init.position.y;
+    init.lightParams.posZ = init.position.z;
+
+    Effect_Add(globalCtx, &sp24, 3, 0, 1, &init);
+    Audio_PlaySoundGeneral(NA_SE_IT_REFLECTION_WOOD, arg2, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+}
+
+#ifdef NON_MATCHING
+// Incomplete, possibly not using the same logic
```

Comment:
Is this equivalent? Because if it's not it shouldn't be under `NON_MATCHING`.

Reply:
If it's now equivalent with the recent improvements, the comment should be updated.